src/partner-access/partner-access.service.ts

```diff
@@ -38,4 +38,8 @@ export class PartnerAccessService {
 
     return await this.partnerAccessRepository.save(partnerAccessDetails);
   }
+
+  async validatePartnerAccessCode(partnerAccessCode: string): Promise<boolean> {
+    return !!(await this.partnerAccessRepository.findOne({ accessCode: partnerAccessCode }));
```

Comment:
Really clean. we should also check for if the PartnerAccess is already associated to a user / is in use. maybe just checking `!! partnerAccess.activatedAt` or `!! partnerAccess.userId`?

I'm also wondering about whether to return specific error messages to the frontend like `ALREADY_IN_USE / INVALID_CODE / DOES_NOT_EXIST` so we can direct the user to help with getting a valid code. One to discuss in standup!